everestrs/everestrs/src/lib.rs

```diff
@@ -394,40 +402,28 @@ pub trait Subscriber: Sync + Send {
     fn on_ready(&self) {}
 }
 
-/// The [Runtime] is the central piece of the bridge between c++ and Rust. We
-/// have to ensure that the `cpp_module` never outlives the [Runtime] object.
-/// This means that the [Runtime] **must** take ownership of `cpp_module`.
+/// The [Runtime] is the central piece of the bridge between c++ and Rust. The
```

Comment:
the old code made sure rust own the c++ module. this was necessary since we passed the refs of the runtime into the c++ module to be called in its callbacks. Now we have a static lifetime of the c++ module which means it for sure outlifes the rust code. Therefore I've changed also the lifetime ouf the rust module to static.

for the subscriber there is a similar logic. We allowed the user to drop the subscriber. This would drop the rust runtime and with the c++ module. Now, if the user drops their subscriber, the weak ref would be empty, triggering a panic. Hence, since everything is now static lifetime, we must forbid that the subscriber goes out of scope